src/libcollections/string.rs

```diff
@@ -1012,6 +1023,15 @@ impl<T: fmt::Show> ToString for T {
     }
 }
 
+#[cfg(not(stage0))]
+impl<T: fmt::String> ToString for T {
+    fn to_string(&self) -> String {
+        let mut buf = Vec::<u8>::new();
+        let _ = fmt::write(&mut buf, format_args!("{}", *self));
+        String::from_utf8(buf).unwrap()
+    }
+}
```

Comment:
the stage0/not(stage0) bits, formatting should now use a `String` instead of a `Vec<u8>`